### DISCUSSION TITLE
MOGA/JEGA source code

### DISCUSSION THREAD

**Discussion by johngriswell** [snl-dakota/dakota#55] [Q&A] [answered]
I am wonderomg if the MOG/JEGA source codes are included or are available somewhere?  Poking around the Daktoa source I found src/JEGAOptimizer.cpp which implements the MOGA alg by interacting with JEGA ... but I can't find the JEGA includes/source that it references.

I am looking to modify the source to used some custom code for more control.

**Comment by briadam** [comment, marked as answer]
You are correct that src/JEGAOptimizer.cpp is the Dakota wrapper that interfaces to the JEGA third-party library. Most of the Dakota TPLs are kept in a separate packages repository, e.g, for JEGA: https://github.com/snl-dakota/dakota-packages/tree/devel/JEGA. All the public Dakota repos can be viewed here: https://github.com/orgs/snl-dakota/repositories

**Reply by tdrayna** [reply]
@briadam 

Hi Brian,

What does this mean for a spack build recipe? I'm trying to update spack to build 6.16 through 6.18 and I see that the cli source releases on github do not have anything in the packages/external directory. Do I need to copy dakota-packages contents into that directory before building?

Thanks,
Travis
